--- configure.in	2006/09/11 09:46:42	1.62
+++ configure.in	2006/09/14 20:34:49	1.63
@@ -33,7 +33,8 @@
 AC_CHECK_LIB(dl, dlopen,          [], AC_MSG_ERROR([*** dl library (libdl) not found]))
 
 dnl libparted
-AC_MSG_CHECKING(for libparted >= 1.7.1)
+LIBPARTED_VERSION=1.7.1
+AC_MSG_CHECKING(for libparted >= $LIBPARTED_VERSION)
 LDFLAGS=-lparted
 AC_TRY_RUN(
 #include <stdio.h>
@@ -41,16 +42,28 @@
 
 int main ()
 {
+	int min_major ;
+	int min_minor ;
+	int min_micro ;
 	int major ;
 	int minor ;
 	int micro ;
 
-	if ( sscanf( ped_get_version(), "%d.%d.%d", &major, &minor, &micro ) == 3 )
-		printf( "Found libparted %s\t", ped_get_version() ) ;
+	if ( sscanf( "$LIBPARTED_VERSION", "%d.%d.%d", &min_major, &min_minor, &min_micro ) == 3 )
+	{
+		if ( sscanf( ped_get_version(), "%d.%d.%d", &major, &minor, &micro ) == 3 )
+		{
+			printf( "Found libparted %s\t", ped_get_version() ) ;
+
+			return ! ((major > min_major) ||
+           			 ((major == min_major) && (minor > min_minor)) ||
+           			 ((major == min_major) && (minor == min_minor) && (micro >= min_micro))) ;
+		}
+	}
 
-        return ! ( major == 1 && minor == 7 && micro >= 1 ) ;
+	return 1 ;
 }
-,AC_MSG_RESULT(OK),AC_MSG_ERROR(*** Requires libparted >= 1.7.1) )
+,AC_MSG_RESULT(OK),AC_MSG_ERROR(*** Requires libparted >= $LIBPARTED_VERSION) )
 
 
 dnl GTKMM
